service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -684,6 +685,18 @@ public ResponseEntity<ApiCloneWorkspaceResult> getCloneWorkspaceResult(
     return new ResponseEntity<>(result, getAsyncResponseCode(result.getJobReport()));
   }
 
+  @Override
+  public ResponseEntity<ApiDataCenterList> listValidDataCenter(UUID workspaceId, String platform) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    workspaceService.validateWorkspaceAndAction(
+        userRequest, workspaceId, SamConstants.SamWorkspaceAction.READ);
```

Comment:
nit: `SamConstants.SamWorkspaceAction` is imported, this can just be `SamWorkspaceAction.READ`